tls/s2n_server_new_session_ticket.c

```diff
@@ -324,6 +324,8 @@ S2N_RESULT s2n_tls13_server_nst_recv(struct s2n_connection *conn, struct s2n_stu
         if (ticket_lifetime == 0) {
             return S2N_RESULT_OK;
         }
+        conn->ticket_lifetime_hint = ticket_lifetime;
```

Comment:
We should probably update the unit tests here as well for this change? 
 https://github.com/aws/s2n-tls/blob/0c4a2d70eae83dd2e66c71945e97a77311b7ccf3/tests/unit/s2n_server_new_session_ticket_test.c#L417-L428